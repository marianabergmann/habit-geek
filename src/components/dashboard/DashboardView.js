import React, { useState, Fragment } from "react";

import HabitTable from "../habits/HabitTable";
import DashboardWeekHeader from "./DashboardWeekHeader";

const DashboardView = ({
  username,
  currentHabits,
  previousHabits,
  currentWeek,
  previousWeek,
  onAddNewHabit,
  onUpdateHabit,
  toggleDayHabit
}) => {
  const [isCurrentWeek, setIsCurrentWeek] = useState(true);

  return (
    <div className="large-size-container">
      <h2 className="mb-3">
        What have you done this week{username ? " " + username : ""}?
      </h2>

      <DashboardWeekHeader
        week={isCurrentWeek ? currentWeek : previousWeek}
        isCurrentWeek={isCurrentWeek}
        hasPreviousWeek={previousWeek}
        toggleWeek={setIsCurrentWeek}
      ></DashboardWeekHeader>

      <HabitTable
        habits={isCurrentWeek ? currentHabits : previousHabits}
        week={isCurrentWeek ? currentWeek : previousWeek}
        toggleDayHabit={toggleDayHabit}
        onUpdateHabit={onUpdateHabit}
      ></HabitTable>

      {isCurrentWeek ? (
        <Fragment>
          <hr />
          <div className="d-flex justify-content-center">
            <button className="btn btn-primary btn-lg" onClick={onAddNewHabit}>
              I'll add a new habit!
            </button>
          </div>
        </Fragment>
      ) : null}
    </div>
  );
};

export default DashboardView;
