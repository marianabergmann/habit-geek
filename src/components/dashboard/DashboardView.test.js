import React from "react";
import { shallow } from "enzyme";
import DashboardView from "./DashboardView";

describe("DashboardView component", () => {
  it("Renders static elements", () => {
    const dashboardView = shallow(
      <DashboardView username="rafa" habits={[{}, {}]} />
    );
    expect(dashboardView).toMatchSnapshot();
  });

  it("Don't show habits if none are stored", () => {
    const dashboardView = shallow(
      <DashboardView username="rafa" habits={[]} />
    );
    expect(dashboardView.find("HabitTable").length).toBe(0);
  });

  it("Don't show username is none is provided", () => {
    const dashboardView = shallow(<DashboardView habits={[]} />);
    expect(dashboardView).toMatchSnapshot();
  });

  it("starts to create a habit", () => {
    const toggleIsCreatingHabitMock = jest.fn();
    const dashboardView = shallow(
      <DashboardView
        username="rafa"
        habits={[]}
        toggleIsCreatingHabit={toggleIsCreatingHabitMock}
      />
    );
    dashboardView.find("button").simulate("click");
    expect(toggleIsCreatingHabitMock).toHaveBeenCalledTimes(1);
  });
});
