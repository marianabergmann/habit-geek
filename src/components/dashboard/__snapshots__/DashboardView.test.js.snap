// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardView component Don't show username is none is provided 1`] = `
<div
  className="large-size-container"
>
  <h2
    className="mb-3"
  >
    What have you done this week
    ?
  </h2>
  <div
    className="d-flex justify-content-center"
  >
    <button
      className="btn btn-primary btn-lg"
    >
      I'll add a new habit!
    </button>
  </div>
</div>
`;

exports[`DashboardView component Renders static elements 1`] = `
<div
  className="large-size-container"
>
  <h2
    className="mb-3"
  >
    What have you done this week
     rafa
    ?
  </h2>
  <HabitTable
    habits={
      Array [
        Object {},
        Object {},
      ]
    }
  />
  <div
    className="d-flex justify-content-center"
  >
    <button
      className="btn btn-primary btn-lg"
    >
      I'll add a new habit!
    </button>
  </div>
</div>
`;
